Change the Users and groups page for Cockpit (#537)
NethServer/docs: administrator-manual/en/accounts.rst

    /var/lib/nethserver/home
    /var/lib/nethserver/vmail
    /var/lib/nethserver/ibay

>     /var/lib/nethserver/nextcloud

Reviewer: I'm not sure that this should be here: nextcloud dir names are based on uuid and it's hard to retrieve the corresponding user.

Author: The section header is "Local accounts provider uninstall". 

I think it requires removing everything under that dir.

Reviewer: No, because admin data are still there.
This is the same for any users not coming from the account providers.

Also the dir store the logs: it should not be deleted manaually

Author: I understand your point. However by your reply I feel there's something odd with ibay and vmail: there we have admin and shared data and users not coming from account providers too!

Back to the list is explanation, in the line above:

    This is the list of system directories containing users and groups data:

It does not state anything about subdir names, so do not consider my previous comment, when I wrote "removing everything". The list is just the place where users and groups data is stored, and removal is not automatic.

I'm still for adding nextcloud to that list, and also any similar directory I may forget.

Reviewer: Ok, fine.

Author: I think we have to fix the whole section because the directory list with users and groups data is the same, no matter the account provider is.

The sentence "A list of users and groups in TSV format is dumped" is true for local providers though.

To configure a remote LDAP accounts provider, go to page :guilabel:`Users & Groups
> Configure the account provider > LDAP > Bind remote LDAP`.

Type the LDAP server IP address in the field :guilabel:`Hostname or IP`. If
the LDAP service runs on a non-standard TCP port, specify it in :guilabel:`TCP
port`. Press the :guilabel:`Check` button to proceed.

Then an LDAP *rootDSE* query is sent to the specified host and a form is filled
with returned data.  Check the values are correct then press the
:guilabel:`Check` button again.

If the LDAP server requires authentication, set :guilabel:`Bind Type` to :guilabel:`Authenticated`.
Set either ``ldaps://`` in :guilabel:`Service URI` or enable :guilabel:`StartTLS` to encrypt the connection.

.. tip::

    If the remote LDAP server is also a |product| installation and
    it is in the LAN (green) network, select :guilabel:`Anonymous bind`

Some applications require an additional configuration step. See also :ref:`dedicated-service-account`.

.. index::
     pair: service; account

.. _dedicated-service-account:

.. _ad-dedicated-service-account:

LDAP account for additional applications
----------------------------------------

> Some additional applications, like *Nextcloud*, *WebTop*, *Roundcube*, *Ejabberd*,
> require read-only access to LDAP services. To be fully operational they
> require a dedicated user account to perform simple LDAP binds.

Reviewer: This a bit repetitive.
What about?

```suggestion
Some additional applications, like *Nextcloud*, *WebTop*, *Roundcube*, *Ejabberd*,
require read-only dedicated user account to perform simple LDAP binds. 
```

For this purpose, the builtin ``ldapservice`` account is automatically created
in local account providers with limited privileges. Its :guilabel:`Bind password` and full
:guilabel:`Bind DN` are shown under :guilabel:`Users & Groups > Account provider > [Details]`.
It is recommended to use those credentials to connect external systems to the account provider.

> On the other hand, if |product| is bound to a remote accounts provider follow these steps:

Reviewer: ```suggestion
On the other hand, if |product| is bound to a remote account provider follow these steps:
```

   credentials in :guilabel:`Users & Groups > Account provider > Edit provider > Authentication credentials for LDAP applications`.

3. If the remote account provider supports TLS, it is recommended to enable the :guilabel:`StartTLS` option or use the ``ldaps://``
   URI scheme in the :guilabel:`Service URI` input field to avoid sending clear-text passwords over the network.

.. warning::

  The |product| AD accounts provider supports TLS. MS-Windows AD might require additional setup to enable TLS.

.. _changing-account-provider-section:

Changing account provider
-------------------------

The configured account provider can be removed by root from
:guilabel:`Users & Groups > Account provider > Change provider`.

When the account provider has been removed, existing files owned
by users and groups must be removed manually. This is
the list of system directories containing users and groups data: ::

    /var/lib/nethserver/home
    /var/lib/nethserver/vmail
    /var/lib/nethserver/ibay
    /var/lib/nethserver/nextcloud

Furthermore, if the account provider is local any user, group and computer

> account is erased.  A list of users and groups in TSV format is dumped to

Reviewer: ```suggestion
account is erased.  A list of users and groups in TSV (Tab Separated Value) format is dumped to
```

----------------------------

To configure a remote LDAP accounts provider, go to page :guilabel:`Users & Groups
> Configure the account provider > LDAP > Bind remote LDAP`.

Type the LDAP server IP address in the field :guilabel:`Hostname or IP`. If
the LDAP service runs on a non-standard TCP port, specify it in :guilabel:`TCP
port`. Press the :guilabel:`Check` button to proceed.

Then an LDAP *rootDSE* query is sent to the specified host and a form is filled
with returned data.  Check the values are correct then press the
:guilabel:`Check` button again.

If the LDAP server requires authentication, set :guilabel:`Bind Type` to :guilabel:`Authenticated`.
Set either ``ldaps://`` in :guilabel:`Service URI` or enable :guilabel:`StartTLS` to encrypt the connection.

.. tip::

    If the remote LDAP server is also a |product| installation and
    it is in the LAN (green) network, select :guilabel:`Anonymous bind`

Some applications require an additional configuration step. See also :ref:`dedicated-service-account`.

.. index::
     pair: service; account

.. _dedicated-service-account:

.. _ad-dedicated-service-account:

LDAP account for additional applications

> ----------------------------------------

Reviewer: I'd move this section to a subsection of "Bind to a remote LDAP server" one by replacing the underline `---` with `^^^`.

Author: Why? That is required for both AD and LDAP providers...

Reviewer: Right, but it's a bit confusing: this is needed only for remote account provides.
Maybe we need to specify it on the title? Or add a note somewhere?

Author: Please review again

Some additional applications, like *Nextcloud*, *WebTop*, *Roundcube*, *Ejabberd*,

> require read-only access to LDAP services. To be fully operational they

Reviewer: ```suggestion
require read-only access to remote LDAP services. To be fully operational they
```

Author: Why _remote_? That's true despite of the account provider thing...

Reviewer: Because for local AD and LDAP everything is auto-configured.

Author: Ok, let's say that explicitly, because it might be not clear that the whole section applies to remote providers only (described in the above sections).

Author: Please review again

require a dedicated user account to perform simple LDAP binds.
                
1. Create a dedicated user account in the remote AD or LDAP provider,
   and set a complex and *non-expiring* password for it.
                
2. Once |product| is successfully bound to a remote AD or LDAP account provider, specify the dedicated user account
   credentials in :guilabel:`Users & Groups > Account provider > Edit provider > Authentication credentials for LDAP applications`.

If the remote account provider supports TLS, it is recommended to enable the :guilabel:`StartTLS` option or use the ``ldaps://``
URI scheme in the :guilabel:`Service URI` input field to avoid sending clear-text passwords over the network.

.. warning::

  The |product| AD accounts provider supports TLS. MS-Windows AD might require additional setup to enable TLS.

Users
=====

If a remote AD or LDAP account provider was configured, the :guilabel:`Users & Groups` page

> can only show lists. It is not possible to modify or delete users and groups from the Server Manager.

Reviewer: ```suggestion
will show read-only lists. It is not possible to modify or delete users and groups from the Server Manager.
```

> On the other hand if a local AD or LDAP account provider was installed, the :guilabel:`Users & Groups` page

Reviewer: ```suggestion
On the other hand, if a local AD or LDAP account provider was installed, the :guilabel:`Users & Groups` page
```
